osu.Game.Tests/Visual/UserInterface/TestSceneSegmentedGraph.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu;
+using osuTK;
+using Vector4 = System.Numerics.Vector4;
+
+namespace osu.Game.Tests.Visual.UserInterface
+{
+    public partial class TestSceneSegmentedGraph : OsuTestScene
+    {
+        private readonly SegmentedGraph<int> graph;
+
+        public TestSceneSegmentedGraph()
+        {
+            Children = new Drawable[]
+            {
+                graph = new SegmentedGraph<int>(10)
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Size = new Vector2(1, 0.5f),
+                },
+            };
+
+            for (int i = 0; i < graph.TierColours.Length; i++)
+            {
+                graph.TierColours[i] =
```

Comment:
This is the only portion I'm not too sure about. It's not common that we expose a `readonly array` for external setting of individual items. This is to avoid oversights like potentially having a single colour updated but the visuals not updating as there's no event flow for the change.

I think I'd prefer the colours to be provided in the constructor, or passed in as the full array (`TierColours = new[] { ... }`) rather than updating one index at a time.